File: client/cpp/gamblerClient.cpp
#include <iostream>
#include "connection.h"
#include "json.hpp"
using json = nlohmann::json;

std::string pwd;
std::string clientName = "cppGamblerClient";
bool gameOver;
bool isStart;
bool gamblerTurn;
int currentWealth;
int currentSlot;
int currentRound;
int totalSlot;//s


void getStatus(){
    std::string ret = sendGet("http://localhost:8080/gambler/status?pwd="+ pwd);
    std::cout<<ret<<std::endl;
    json j = json::parse(ret);
    totalSlot = j["totalSlot"];
    gamblerTurn = j["gamblerTurn"];
    currentSlot = j["currentSlot"];
    isStart = j["start"];
    gameOver =j["gameOver"];
    currentRound = j["currentRound"];
    currentWealth = j["deposit"];
}

void sendMove(int slot,int bet){
    assert(slot<=totalSlot);
    assert(bet>=1&&bet<=3);
    std::string ret = sendPost("http://localhost:8080/gambler/sendMove?pwd="+pwd+"&slot="+ std::to_string(slot)+"&bet="+std::to_string(bet));
    std::cout<<ret<<std::endl;
    json j = json::parse(ret);
    totalSlot = j["totalSlot"];
    gamblerTurn = j["gamblerTurn"];
    currentSlot = j["currentSlot"];
    isStart = j["start"];
    gameOver =j["gameOver"];
    currentRound = j["currentRound"];
    currentWealth = j["deposit"];

}
int main() {
    initCurl();
    pwd=sendPost("http://localhost:8080/gambler/join?name="+ clientName);
    getStatus();
    while(!isStart){
        getStatus();
    }

    while(!gameOver){
        if(gamblerTurn){
            //TODO: change the strategy
           sendMove(1,std::min(3,currentWealth));
        }
        getStatus();
    }
    cleanCurl();
    return 0;
}

File: client/python/gamblerClient.py
import json
import requests

URL = "http://localhost:8080"
name = "pythonGamblerClient"


def sendPost(path):
    r = requests.post(URL + path)
    return r.text


def sendGet(path):
    r = requests.get(URL + path)
    return r.text


def getStatus(pwd):
    ret = sendGet("/gambler/status?pwd=" + pwd)
    return json.loads(ret)


def sendMove(pwd, slot, bet):
    ret = sendPost("/gambler/sendMove?pwd=" + pwd + "&slot=" + str(slot) + "&bet=" + str(bet))
    return json.loads(ret)


if __name__ == '__main__':
    pwd = sendPost("/gambler/join?name=" + name)
    status = getStatus(pwd)
    print(status)
    while not status["start"]:
        status = getStatus(pwd)

    while not status["gameOver"]:
        if status["gamblerTurn"]:
            # TODO: set current slot and bet
            sendMove(pwd, 2, min(3, status["deposit"]))
        status = getStatus(pwd)
